### PR TITLE
Warnings and specialized diffs when switching between sensitive values

### PR DESCRIPTION
**Pull request by pselle**
This PR adds a warning that will display in the diff output if a value is changing its sensitivity. It is conservative, in that if the sensitivity is changing (either to _or_ from sensitive) it will not display the value, but will show a `(sensitive)`.

Some examples of this in action:

Changing from one sensitive value to another sensitive value, the `(sensitive)` remains in place and no warning is displayed:
Screen Shot 2020-09-23 at 4 14 52 PM

Changing from a sensitive value to a non sensitive value, a warning is displayed in the UI:
Screen Shot 2020-09-23 at 4 15 30 PM
Same thing, but showing this appearing more than once in a plan:
Screen Shot 2020-09-23 at 4 16 01 PM
When a value will no longer be sensitive, this warning displays:
Screen Shot 2020-09-23 at 4 17 06 PM

This PR adds coverage for this specialized diff between all primitive types, as more work is needed (both in `diff.go` but also more widely in Terraform) on complex values and sensitivity, so I indend to do that in a separate PR so we can discuss the design of "how we show sensitivity" here. 

Edit to add: There is some dealing with map types already in here, but it is not exhaustive.

### CODE REVIEW

File: command/format/diff.go
Comment (alisdair) [inline command/format/diff.go:784]: With the caveat that I don't think this is important (and may be wrong), I have a small suggestion. `ContainsMarked` is relatively expensive, and we can end up calling it several times in this function. Would the following sketch make sense?

```go
unmarkedOld, oldMarks := old.UnmarkDeep()
unmarkedNew, newMarks := new.UnmarkDeep()

// …

if len(oldMarks) > 0 {
    p.buf.WriteString("(sensitive)")
    return
}
```

File: command/format/diff.go
Comment (alisdair) [inline command/format/diff.go:1121]: I think this will panic if the map itself is marked.

**Review by alisdair — APPROVED** [review]
I really like this UI design! I have some minor thoughts inline that needn't be addressed before merge, so ✅ from me.

I think we might have to add tests and make changes if we add support for marking collections/structural types as sensitives, but there's no need for that in this PR.

File: command/format/diff.go
Comment (pselle) [reply command/format/diff.go:784]: I like it! Thanks for the suggestion!

File: command/format/diff.go
Comment (pselle) [reply command/format/diff.go:1115]: It does! This PR does not deal with non-primitive values.

**Review by apparentlymart — APPROVED** [review]
This looks good!

I have a light preference towards only making the string "Warning: " be yellow and letting the rest be the default color, like we do in our diagnostic messages. That was motivated by seeing that it's somewhat common for folks to have terminal color schemes where certain colors are not readable against the background -- sometimes a light-colored background where yellow isn't contrasty enough, and other times just the general oddness of the Windows command prompt's color handling.

However, that's a minor detail and not a strong preference because we can always iterate these details based on feedback, so I'll leave the final decision up to you.

**Comment by pkolyvas** [comment]
Looks great and the additional guidance on change(s) is a really helpful touch.

**Pull request merged**
